### PR TITLE
Richb hanover patch 1

### PR DESCRIPTION
**Pull request by richb-hanover**
Full editorial pass across the whole document. I'll also send a Word version with Track Changes on...

### CODE REVIEW

**Comment by cpaasch** [comment]
Hi Rich! A bunch of unrelated files were added with this PR. Can you remove these? (going over your change in .md now)

**Comment by richb-hanover** [comment]
I have found these tools (markdownlint and spellcheck) really useful for maintaining other large Markdown documents. They found about 20 misspellings and other problems in the document.

But I don't really understand the workflow for writing/maintaining an RFC for an I-D. Will they hinder the creation of the spec? Thanks.

**Comment by richb-hanover** [comment]
Done. It's easy to add them back.

File: draft-cpaasch-ippm-responsiveness.md [hunk 1]
```diff
@@ -59,213 +58,180 @@ informative:
...
-extend of bufferbloat.
+of latency-spikes. An example of this is lag occurring
+during a video-conference, where a connection is shown as unstable.
+
+Since bufferbloat can cause short-lived disruptions of the user experience,
+it is hard to narrow down the cause and make the user aware of it.
+Popular measurement platforms that focus on speed and idle latency only obscure the bufferbloat problem.
+
+Finally, it is important to remember that buffers only fill behind the slowest "bottleneck" link of the network.
+The "bottleneck node" is one where the queueing occurs during heavy traffic.
```
Comment (cpaasch) [inline draft-cpaasch-ippm-responsiveness.md:108]: I'm not sure about this phrase here. What does it add to the message just above where we try to conveye why bufferbloat is a subtle problem that on one side largely impacts user-experience while on the other side is so short-lived that it gets often just shrugged away.

File: draft-cpaasch-ippm-responsiveness.md [hunk 2]
```diff
@@ -59,213 +58,180 @@ informative:
...
+
+We believe that it is necessary to create a standardized way for measuring the extent of bufferbloat
+in a network and express it in a comprehensible way.
+Existing network measurement tools could add a responsiveness measurement to their set of metrics.
+It will also raise the awareness to the problem and shift the focus away from simply quantifying
+network quality with the two "standard: measures: throughput and idle latency.
```
Comment (cpaasch) [inline draft-cpaasch-ippm-responsiveness.md:114]: There must be a typo with ```"standard: measures: throughput```

File: draft-cpaasch-ippm-responsiveness.md [hunk 3]
```diff
@@ -59,213 +58,180 @@ informative:
...
-   time-frame and short being anything below 20 seconds.
-
-# Measuring Responsiveness
+    As a side-note: other types of traffic are gaining in popularity (HTTP/3)
+and/or are already being used widely (RTP).
+Due to traffic prioritization and QoS rules on the Internet, each of these may experience
+completely different path-characteristics and should also be measured separately.
+
+2. The Internet is marked by the deployment of countless middleboxes like transparent TCP proxies or traffic prioritization for certain types of traffic.
+The measurement methodology should explore/exploit all of these as they affect the user- experience.
```
Comment (cpaasch) [inline draft-cpaasch-ippm-responsiveness.md:151]: Not sure about the "exploit" term here. Can you elaborate?

File: draft-cpaasch-ippm-responsiveness.md [hunk 4]
```diff
@@ -59,213 +58,180 @@ informative:
...
 
-Measuring in parallel allows to achieve higher overall confidence in the
+Measuring in parallel achieves higher overall confidence in the
 results within the time constraint of the entire test. For example, if the
 overall time constraint for the test is 20 seconds, running uplink and downlink
 sequentially would allow for only 10 seconds of test per direction, while
-parallel measurement will allow for 20 seconds of testing in both directions.
+parallel measurement provides 20 seconds of testing in both directions.
 
 However, a number caveats come with measuring in parallel:
```
Comment (cpaasch) [inline draft-cpaasch-ippm-responsiveness.md:217]: While you are at it, can you add an "of" between "number" and "caveats" ? ;-)

File: draft-cpaasch-ippm-responsiveness.md [hunk 5]
```diff
@@ -389,36 +351,40 @@ responsiveness under working conditions.
 ### Statistical Confidence
 
 It remains an open question as to how many repetitions of the above described
-latency measurements a tool would need to execute. One could imagine a
+latency measurements a tool would need to execute to gain statistical confidence. One could imagine a
 computation of the variance and confidence interval that would drive the number
 of measurements and balance the accuracy with the speed of the measurement
 itself.
 
+This is an open area for investigation. *(true?)*
```
Comment (cpaasch) [inline draft-cpaasch-ippm-responsiveness.md:359]: Yes, this is true :) You can remove the ```*(true?)*``` :)

**Review by cpaasch — CHANGES_REQUESTED** [review]
First pass done! Looks good besides some of the comments I added.

**Comment by cpaasch** [comment]
One more comment: Feel free to add yourself in the Acknowledgements section :-) (otherwise, I will do it ;-) )

**Comment by richb-hanover** [comment]
Thank you for all the comments. I will review them and roll them into a future update. (By Friday?)

**Comment by richb-hanover** [comment]
PS Two scaffolding questions:

* How do you build kramdown? brew doesn't seem to do it, nor did I find a docker image. (I am using https://xml2rfc.tools.ietf.org/experimental.html to check the RFC, but it's awkward...)
* Those extra files I initially added into my patch make it easier to maintain a well-formatted Markdown document and to check the spelling. Are they not useful? 

Thanks again.

**Comment by cpaasch** [comment]
Hello,

> PS Two scaffolding questions:
> 
> * How do you build kramdown? brew doesn't seem to do it, nor did I find a docker image. (I am using https://xml2rfc.tools.ietf.org/experimental.html to check the RFC, but it's awkward...)

Yes, that's the one you should use. When you have all the tools installed, you should be able to just type ```make```.

> * Those extra files I initially added into my patch make it easier to maintain a well-formatted Markdown document and to check the spelling. Are they not useful?

How can one use them? Which commands do you run? If it's useful, we can integrate that in the makefile.


Thanks,
Christoph

**Comment by richb-hanover** [comment]
Hi Christoph - I updated the document with an eye toward it being cited by my note to the IAB Workshop next week. Thanks.

**Comment by cpaasch** [comment]
@richb-hanover  - I'm going to merge this as is and do a pass over it to fix some of the comments I left. IAB camera-ready deadline is today, so we need to move quickly.

Nevertheless, we can still evolve the document beyond the IAB workshop as ultimately we want this to become a full IETF-standard.

**Comment by richb-hanover** [comment]
> I'm going to merge this as is and do a pass over it to fix some of the comments I left. IAB camera-ready deadline is today, so we need to move quickly.
> 
> Nevertheless, we can still evolve the document beyond the IAB workshop as ultimately we want this to become a full IETF-standard.

Absolutely. Thanks!

**Comment by richb-hanover** [comment]
PS I created a new PR that incorporates the PDF, HTML, text, PS, and XML versions so that a more readable form is available without running `make`

**Pull request merged**